File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/typescript/F08.P2.ts
function probe(a: number, b: number): number {
// BEGIN PROBE F08.P2
const q = b === 0 ? 0 : Math.trunc(a / b)
return q
// END PROBE F08.P2
}

console.log(probe(-7, 0), probe(-7, 2))

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/zig/F08.P2.zig
const std = @import("std");

var src: i32 = 0;

fn probe() i32 {
    const a: i32 = 7;
    const b: i32 = src;
// BEGIN PROBE F08.P2
const q = std.math.divTrunc(i32, a, b) catch 0;
return q;
// END PROBE F08.P2
}

pub fn main() !void {
    var threaded: std.Io.Threaded = .init_single_threaded;
    const io = threaded.io();
    var out_buf: [64]u8 = undefined;
    var out = std.Io.File.stdout().writer(io, &out_buf);
    try out.interface.print("{d}\n", .{probe()});
    try out.interface.flush();
}
